buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/UpdateAgpVersions.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.gradlebuild
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import org.gradle.build.ReproduciblePropertiesWriter
+import org.jsoup.Jsoup
+import java.util.Properties
+
+
+/**
+ * Fetch the latest AGP versions and write a properties file.
+ * Never up-to-date, non-cacheable.
+ */
+abstract class UpdateAgpVersions : DefaultTask() {
+
+    @get:Internal
+    abstract val comment: Property<String>
+
+    @get:Internal
+    abstract val minimumSupportedMinor: Property<String>
+
+    @get:Internal
+    abstract val propertiesFile: RegularFileProperty
+
+    @TaskAction
+    fun fetch() {
+
+        val minimumSupported = minimumSupportedMinor.get()
```

Comment:
❌ I would extract two methods from `fetch` to make it more obvious: `latest(): List<String>` and `nightly(): String`

Reply:
Good point. I did that and also changed the implementation to only rely on maven metadata instead of a 3rd party site.